### PR TITLE
Added alert when saving frozen sandbox

### PR DESCRIPTION
**Pull request by JonShort**
**What kind of change does this PR introduce?**

This PR adds a browser alert prior to saving a frozen sandbox. The alert describes that the sandbox will be forked.

Also included is a minor code spelling fix (FreezeConatainer > FreezeContainer)

**What is the current behavior?**

Currently saving a frozen sandbox will silently fork the sandbox, similar to saving a sandbox your don't own.

Issue #1212 mentioned that a prompt identifying that the sandbox will be forked could be useful.

**What is the new behavior?**

When saving edits to a frozen sandbox which you own, an alert will appear describing that the sandbox will be forked.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table 

**Comments / Questions**

Q: Could this be a `confirm()` rather than an `alert()`?

A: This would be my ideal implementation, allowing the user to cancel and 'unfreeze' their sandbox if they wish. Unfortunately I don't know a way to cancel the proceeding cerebral `sequence` (which would need to be approached generically as save is triggered in many places). I'd be interested in seeing other approaches to this problem which allow this functionality.

Q: Does this negatively affect UX?

A: As mentioned in the issue prompting this work, the application silently forking a frozen sandbox felt like a bit of a dark pattern to the individual raising the issue. I'm happy to offer up this PR, and get feedback around the UX impact of it.

If anyone can think of a message clearer than "This sandbox is frozen, and will be forked." please let me know, I feel the message might not be clear to someone who doesn't know the concept of 'frozen' and 'forked' sandboxes 😃

### CODE REVIEW

File: packages/app/src/app/store/sequences.js [hunk 1]
```diff
@@ -146,15 +147,25 @@ export const forkSandbox = sequence('forkSandbox', [
   },
 ]);
 
+export const forkFrozenSandbox = sequence('forkFrozenSandbox', [
+  when(state`editor.currentSandbox.isFrozen`),
+  {
+    true: [alertForkingFrozenSandbox, forkSandbox],
```
Comment (CompuIves) [inline packages/app/src/app/store/sequences.js]: To make this work with a `confirm` we can probably use the `path` directive of Cerebral. Here is an example: https://github.com/CompuIves/codesandbox-client/blob/master/packages/app/src/app/store/actions.js#L8-L19

The idea is that you can return different paths from an action, and in the sequence you can define what to do for every returned path. So here you could do:

```
{
	confirm: [forkSandbox],
    cancel: []
}
```

I think that would solve your question for how to use `confirm`.

File: packages/app/src/app/store/modules/editor/actions.js [hunk 2]
```diff
@@ -301,6 +301,10 @@ export function confirmForkingOwnSandbox({ browser, path }) {
     : path.cancelled();
 }
 
+export function alertForkingFrozenSandbox({ browser }) {
```
Comment (CompuIves) [inline packages/app/src/app/store/modules/editor/actions.js]: So here you could add `path` and return either `path.confirm`/`path.cancel`.

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Project/elements.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const PrivacyContainer = styled.span`
   margin-bottom: 1rem;
 `;
 
-export const FreezeConatainer = styled.span`
+export const FreezeContainer = styled.span`
```
Comment (CompuIves) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/Project/elements.js]: Nice find!

File: packages/app/src/app/store/sequences.js (hunk 1 above)
Comment (JonShort) [reply packages/app/src/app/store/sequences.js]: I've given this a try locally, and the fork is cancelled as we'd expect when user cancels the confirm 🎉

However the issue then becomes cancelling the remaining actions e.g:
image

Here when `ensureOwnedEditable` returns without forking, the actions below are still dispatched (e.g. save is performed on frozen sandbox).

To be honest I feel as though it'd be too disruptive to the architecture to add in `if` `else` type logic, especially in this PR 😆

As this PR is a UX enhancement, i'll leave it as-is, feel free to reject/approve as you see fit 👍

**Comment by karlhorky** [comment]
This seems like a cool change, just wanted this exact feature.

Is there anything still open before this can be merged?

**Comment by JonShort** [comment]
I think this PR is perhaps a bit too old now, I'll close it so others can tackle this issue if they wish 👍

**Comment by SaraVieira** [comment]
For bug week I will finish your work!

Thank you so much! I will make sure to add you as a co-author and contributor

**Comment by karlhorky** [comment]
Work continuing in #1716 for those who want to follow along :)

**Pull request merged**